src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -166,7 +167,12 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     codeWriter.EndBlock();
                 }
 
-                return stringWriter.ToString();
+                var verbs = endpoints
+                    .Select(endpoint => endpoint.EmitterContext.HttpMethod!)
+                    .Where(verb => verb is not null)
+                    .ToImmutableHashSet();
+
+                return (stringWriter.ToString(), verbs);
```

Comment:
Can we process the verbs in a separate incremental pipeline?

```
var helperFields = endpoints.Select(endpoint => ...);
```

We might even be able to implement a custom comparer in `WithComparer` so we only pick up unique instances of the HTTP method. Thoughts?

Reply:
I'll take a look 👍